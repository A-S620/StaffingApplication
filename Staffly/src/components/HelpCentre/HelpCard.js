import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import {Card, CardContent, Grid, Typography} from '@material-ui/core'
import ClearIcon from '@material-ui/icons/Clear';
import {connect} from "react-redux";
import IconButton from "@material-ui/core/IconButton";
import {store} from "../../store/Store";

const useStyles = makeStyles({
    root: {
        minWidth: 475,
        margin: "10px",
    },
    bullet: {
        display: 'inline-block',
        margin: '0 2px',
        transform: 'scale(0.8)',
    },
    title: {
        fontSize: 14,
    },
    pos: {
        marginBottom: 12,
    },

});

function HelpCard(props) {
    const classes = useStyles();
    const bull = <span className={classes.bullet}>â€¢</span>;
    const [cardData, setCardData] = React.useState(props.notifs);
    // setCardData(props);
    // console.log(cardData[0].NotifId);

    // {props.notifs.map((singleNotif) => {
    //
    //     return (
    //         cardData.push(singleNotif)
    //     )
    //
    // })}
    const handleDelete = (cardToDelete) => () => {
        console.log(cardToDelete.NotifId);
        var index= cardData.indexOf(cardToDelete);
        console.log(index);
        cardData.splice(index,1);
        // setCardData((cards) => cards.filter((card) => card.NotifId !== cardToDelete.NotifId));
    };
    return (

        <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
        >

            {props.help.map((data) =>{
                return(
                    <Card className={classes.root}>
                        {/*<Grid*/}
                        {/*    container*/}
                        {/*    direction="row"*/}
                        {/*    justify="flex-end"*/}
                        {/*    alignItems="flex-start"*/}
                        {/*>*/}
                        {/*    <IconButton*/}
                        {/*        onClick={ () =>{*/}
                        {/*            store.dispatch({*/}
                        {/*                type: "DELETE_NOTIFICATION",*/}
                        {/*                payload: data.NotifId*/}
                        {/*            })*/}
                        {/*        }*/}

                        {/*        }*/}
                        {/*    >*/}
                        {/*        <ClearIcon/>*/}
                        {/*    </IconButton>*/}

                        {/*</Grid>*/}
                        <CardContent>
                            <Typography variant="h5" component="h2">
                                {data.PostTitle}
                            </Typography>
                            <Typography className={classes.pos} color="textSecondary">
                                {data.PostDate}
                            </Typography>
                            <Typography variant="body2" component="p">
                                {data.PostText}
                            </Typography>
                        </CardContent>

                    </Card>
                )
            })}



        </Grid>
    );
}

const mapStateToProps = state => {
    return {
        help: state.help
    }
}
export default connect(mapStateToProps, {})(HelpCard);