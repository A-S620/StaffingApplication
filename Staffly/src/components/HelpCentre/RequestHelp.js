import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import Slide from '@material-ui/core/Slide';
import CssBaseline from "@material-ui/core/CssBaseline";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import {makeStyles} from '@material-ui/core/styles';
import IconButton from "@material-ui/core/IconButton";
import ClearIcon from "@material-ui/icons/Clear";

const useStyles = makeStyles((theme) => ({
    paper: {
        marginTop: theme.spacing(2),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    avatar: {
        margin: theme.spacing(1),
        backgroundColor: theme.palette.secondary.main,
    },
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: theme.spacing(3),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    },
    button: {
        margin: "5px",
    }
}));
const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
});

export default function RequestHelp(props) {
    const classes = useStyles();
    const [open, setOpen] = React.useState(false);
    const [row, setRow] = React.useState(props.selectedRow);
    const [title, setTitle] = React.useState();
    const [comment, setComment] = React.useState();


    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };



    // console.log(props.selectedRow.ShiftStartDate);

    return (
        <div>
            <Button className={classes.button} variant="outlined" color="primary" onClick={() => {
                handleClickOpen();
            }
            }>
                Request Help
            </Button>
            <Dialog
                open={open}
                TransitionComponent={Transition}
                keepMounted
                onClose={handleClose}
            ><Grid
                container
                direction="row"
                justify="flex-end"
                alignItems="flex-start"
            >
                <IconButton onClick={handleClose}>
                    <ClearIcon/>
                </IconButton>

            </Grid>
                <DialogContent>
                    <DialogContentText>
                        <Container component="main" maxWidth="xs">
                            <CssBaseline/>
                            <div className={classes.paper}>
                                <Typography component="h1" variant="h5">
                                    Help Request
                                </Typography>

                                <form className={classes.form} noValidate>


                                    <Grid container spacing={2}>


                                        <Grid item xs={12}>

                                            <TextField
                                                variant="outlined"
                                                required
                                                fullWidth
                                                id="title"
                                                label="Title"
                                                name="title"
                                                autoComplete="title"
                                                onChange={(event) => {
                                                    setTitle(event.target.value)
                                                }}
                                            />

                                        </Grid>
                                        <Grid item xs={12}>
                                            <TextField
                                                style={{width :400}}
                                                required
                                                fullwidth
                                                id="Comment"
                                                label="Comment"
                                                multiline
                                                rows={6}
                                                variant="outlined"
                                                value={comment}
                                                onChange={(event) => {
                                                    setComment(event.target.value);

                                                }}
                                            />
                                        </Grid>



                                    </Grid>

                                </form>
                            </div>
                        </Container>
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button
                        type="submit"
                        fullWidth
                        variant="outlined"
                        color="primary"
                        className={classes.submit}

                    >
                        Submit Request
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );

}

