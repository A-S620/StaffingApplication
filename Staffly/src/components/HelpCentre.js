import React, { Component } from 'react';
import HelpCard from "./HelpCentre/HelpCard";
import RequestHelp from "./HelpCentre/RequestHelp";


class HelpCentre extends Component {
    render() {
        return (
            <div className="HelpCentre" style={{marginTop:60}} >
                <div>
                    <RequestHelp/>
                    <HelpCard/>
                </div>
            </div>
        );
    }
}

export default HelpCentre;