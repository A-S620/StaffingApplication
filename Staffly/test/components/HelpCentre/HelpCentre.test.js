import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux'
import HelpCentre from "../../../src/components/HelpCentre";
describe('App ', () => {
    it('should render correctly ', () => {
        const component = shallow(<HelpCentre  />);

        expect(component).toMatchSnapshot();
    });
});