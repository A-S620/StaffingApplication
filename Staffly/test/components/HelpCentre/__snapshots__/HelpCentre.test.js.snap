// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  should render correctly  1`] = `
<div
  className="HelpCentre"
  style={
    Object {
      "marginTop": 60,
    }
  }
>
  <div>
    <Connect(HelpCard) />
  </div>
</div>
`;
